### PR TITLE
Feature 1167/refactor template backend

### PR DESCRIPTION
**Pull request by smithj11**
Template backend refactored to connect templates with template questions. It should be ready to perform all template-based operations, including saving w or w/o questions, editing template details and/or editing, deleting, or changing the order of questions, deleting template and all associated questions.

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplateRepository.java [hunk 1]
```diff
@@ -25,10 +24,6 @@
 
     @Override
     <S extends FeedbackTemplate> S save(@Valid @NotNull @NonNull S entity);
-
-    @Query(value = "UPDATE feedback_templates SET active = false WHERE id = CAST(:id as varchar)")
-    Optional<FeedbackTemplate> softDeleteById(@NotNull UUID id);
-
     @Override
     <S extends FeedbackTemplate> S update(@Valid @NotNull @NonNull S entity);
```
Comment (JEdwardBrown) [inline server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplateRepository.java:28]: Are both @NotNull and @NonNull needed here?

**Review by JEdwardBrown — APPROVED** [review]

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestController.java [hunk 2]
```diff
@@ -79,11 +79,9 @@ public FeedbackRequestController(FeedbackRequestServices feedbackReqServices,
...
      */
     @Delete("/{id}")
-    public Single<HttpResponse> delete(@NotNull UUID id) {
-        return Single.fromCallable(() -> feedbackReqServices.delete(id))
-                .observeOn(Schedulers.from(eventLoopGroup))
-                .map(successFlag -> (HttpResponse) HttpResponse.ok())
-                .subscribeOn(Schedulers.from(executorService));
+    public HttpResponse<?> delete(@NotNull UUID id) {
+        feedbackReqServices.delete(id);
+        return HttpResponse.ok();
```
Comment (mkimberlin) [inline server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestController.java:84]: Why are you removing the reactive bits? This has a significant effect and isn't advisable. Were you running into problems?

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplate.java [hunk 3]
```diff
@@ -39,49 +34,66 @@
     @Column(name = "description")
     @Nullable
     @TypeDef(type = DataType.STRING)
-    @Schema(description = "description of feedback template", required = false)
+    @Schema(description = "description of feedback template")
     private String description;
 
-    @Column(name = "createdBy")
+    @Column(name = "creatorId")
```
Comment (mkimberlin) [inline server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplate.java:40]: go lowercase with these, please or you'll run into issues with postgres

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplateController.java [hunk 4]
```diff
@@ -78,14 +78,10 @@ public FeedbackTemplateController(FeedbackTemplateServices feedbackTemplateServi
...
-    public Single<HttpResponse<FeedbackTemplateResponseDTO>> delete(@NotNull UUID id) {
-        return Single.fromCallable(() -> feedbackTemplateServices.delete(id))
-                .observeOn(Schedulers.from(eventLoopGroup))
-                .map(savedFeedbackTemplate -> (HttpResponse<FeedbackTemplateResponseDTO>) HttpResponse
-                        .ok()
-                        .headers(headers -> headers.location(URI.create("/feedback_template/" + savedFeedbackTemplate.getId())))
-                        .body(fromEntity(savedFeedbackTemplate)))
-                .subscribeOn(Schedulers.from(executorService));
+    public HttpResponse<?> delete(@NotNull UUID id) {
+        feedbackTemplateServices.delete(id);
```
Comment (mkimberlin) [inline server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplateController.java:82]: Same issue here...we can't remove the reactive bits or we block the application from servicing other requests until this request is finished.

**Comment by pieperm** [comment]
The delete endpoints should now be non-blocking with the most recent commit. It was changed because a large majority of delete endpoints do not use any sort of scheduling.

**Comment by pieperm** [comment]
Column names have been updated to use snake case 🐍

File: server/src/main/resources/db/common/V73__alter_column_naming_convention.sql [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+RENAME COLUMN requestId to request_id;
+
+ALTER TABLE feedback_request_questions
+RENAME COLUMN questionContent TO question;
+
+ALTER TABLE feedback_request_questions
+RENAME COLUMN orderNum TO question_number;
+
+
+-- Rename columns in table feedback_templates
```
Comment (bhatiam12) [inline server/src/main/resources/db/common/V73__alter_column_naming_convention.sql:37]: can we include V72 migrations in here so all feedback_templates are together ? In that case you will not need V72

Comment (pieperm) [reply server/src/main/resources/db/common/V73__alter_column_naming_convention.sql]: Good idea, the migrations have been merged together into V72 in commit `e8c6e66`

**Comment by bhatiam12** [comment]
@smithj11 @pieperm This branch has conflicts.

**Review by bhatiam12 — APPROVED** [review]

**Pull request merged**